content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,342 @@
+command. In our case we can use `create_workflow_from_json` function
+where we need to define all the needed parameters:
+
+```python
+from reana_client.api.client import create_workflow_from_json
+
+create_workflow_from_json(
+    workflow_json=my_workflow,
+    name=my_workflow_name,
+    access_token=my_reana_token,
+    parameters=my_inputs,
+    workflow_engine=my_workflow_type)
+```
+
+Output:
+
+```python
+{
+  'message': 'Workflow workspace created',
+  'workflow_id': '6cd613eb-f2fb-411b-9601-c89599925759',
+  'workflow_name': 'root6-roofit'
+}
+```
+
+### Uploading files
+
+After creating our workflow we also need to upload all the input files to
+the workflow workspace. Same functionality can be achieved with REANA client
+[upload](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-upload)
+command. It can be done as follows:
+
+```python
+from reana_client.api.client import upload_to_server
+
+abs_path_to_input_files = [os.path.abspath(f) for f in my_inputs['files']]
+upload_to_server(my_workflow_name, abs_path_to_input_files, my_reana_token)
+```
+
+### Starting workflow
+
+At this point we have done all what is needed to start our workflow in REANA.
+To do it with REANA client we could call [start](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-start)
+command. But in this case it can be started by simply calling `start_workflow`
+function and defining workflow name and access token. Additionally, we
+can also override workflow parameters while starting the workflow, but let's
+keep the original ones for now:
+
+```python
+from reana_client.api.client import start_workflow
+
+start_workflow(my_workflow_name, my_reana_token, {})
+```
+
+Output:
+
+```python
+{
+  'message': 'Workflow submitted.',
+  'run_number': 1,
+  'status': 'queued',
+  'user': '1d27737b-5b80-4da2-87bb-816a20e110bd',
+  'workflow_id': '6cd613eb-f2fb-411b-9601-c89599925759',
+  'workflow_name': 'root6-roofit'
+}
+```
+
+### Checking workflow status
+
+After starting our workflow we might want to check its current status.
+To inspect the workflow status using REANA client we could use [status](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-status)
+command. Similarly, it can be achieved with `get_workflow_status` function.
+Since it will take couple of seconds for all the computations to be finished
+while executing our workflow, we will use a loop to wait and regularly check
+what is the current status of our workflow:
+
+```python
+import time
+from reana_client.api.client import get_workflow_status
+
+while True:
+    status_details = get_workflow_status(my_workflow_name, my_reana_token)
+    print('Current status: ', status_details['status'])
+    if status_details['status'] == 'finished':
+        break
+    time.sleep(1)
+```
+
+Output:
+
+```python
+Current status:  running
+...
+Current status:  finished
+```
+
+### Checking workflow logs
```

Comment:
When our workflow finished, or when we would like to debug and investigate how the workflow run, we may want to display the logs of the concrete computational steps. In command-line interface, we could use the `logs` CLI command. In Python, the corresponding function to use is `get_workflow_logs()` function. We need to specify only the workflow name and the access token. Optionally, we can add parameters such as list of steps the logs of which we are interested to see, or pagination parameters in case the log output is too large or too verbose. As an example, let us retrieve the last 20 lines of the "fitdata" step:

- FIXME or keep the overall "undisplayable" function?

...

Example output:

...